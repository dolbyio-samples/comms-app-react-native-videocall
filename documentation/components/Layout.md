# Layout

The Layout component is a wrapper responsible for displaying a full-screen view of a conference and setting a background color.

## Props

| Name                 | Type                    | Default | Description                                               |
| -------------------- | ----------------------- | ------- | --------------------------------------------------------- |
| `children`           | ReactNode               | -       | The content of the component.                             |
| `backgroundColor`?   | ColorKey                | -       | The background color of the layout.                       |
| `testID` ?           | string                  | -       | The unique E2E test handler.                              |

## Examples

### React Native

```javascript
return <Layout>....</Layout>;
```
