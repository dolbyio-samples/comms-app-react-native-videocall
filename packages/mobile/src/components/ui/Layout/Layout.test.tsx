import { defaultTheme as theme } from '@dolbyio/comms-uikit-common';
import React from 'react';
import { Text } from 'react-native';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';

import { render, mergeStyles } from '../../../utils/test-utils';

import Layout from './Layout';

const { colors } = theme;

const testID = 'testID';
const customComponent = <Text testID="say-hello">Hello World</Text>;
const bgColor = 'grey.800';

describe('Layout component', () => {
  test('Passes TestID', () => {
    const { getByTestId } = render(<Layout testID={testID}>Hello World</Layout>);
    expect(getByTestId(testID)).not.toBeNull();
  });

  test('Should render children correctly', () => {
    const { getByTestId } = render(<Layout testID={testID}>{customComponent}</Layout>);
    const element = getByTestId('say-hello');
    expect(element).not.toBeNull();
    expect(element).toHaveTextContent('Hello World');
    // default backgroundColor valiation
    const layoutElm = getByTestId(testID);
    const styles = mergeStyles(layoutElm.props.style);
    expect(styles.backgroundColor).toBe(colors.background);
  });

  test('Should use given backgroundColor', () => {
    const { getByTestId } = render(
      <Layout backgroundColor={bgColor} testID={testID}>
        {customComponent}
      </Layout>,
    );
    const layoutElm = getByTestId(testID);
    const styles = mergeStyles(layoutElm.props.style);
    expect(styles.backgroundColor).toBe(colors.grey[800]);
  });
});

describe('Layout component snapshot', () => {
  test('should match default param configuration', () => {
    const tree = renderer.create(<Layout testID={testID}>{customComponent}</Layout>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match with given backgroundColor', () => {
    const tree = renderer
      .create(
        <Layout backgroundColor={bgColor} testID={testID}>
          {customComponent}
        </Layout>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
