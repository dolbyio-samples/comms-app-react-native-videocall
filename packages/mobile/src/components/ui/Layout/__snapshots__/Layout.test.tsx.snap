// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component snapshot should match default param configuration 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      },
      Object {
        "backgroundColor": "#14141A",
      },
    ]
  }
  testID="testID"
>
  <Text
    testID="say-hello"
  >
    Hello World
  </Text>
</View>
`;

exports[`Layout component snapshot should match with given backgroundColor 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      },
      Object {
        "backgroundColor": "#292930",
      },
    ]
  }
  testID="testID"
>
  <Text
    testID="say-hello"
  >
    Hello World
  </Text>
</View>
`;
